src/dcp/dcp.c

```diff
@@ -346,6 +359,291 @@ static int daos_any_error(int rank, bool local_daos_error, int flag_daos_args)
+        uint32_t number = 5;
+        rc = daos_obj_list_recx(*src_oh, DAOS_TX_NONE, dkey, akey,
+	                        &size, &number, recxs, eprs,
+				&recx_anchor, true, NULL);
+
+        /* if no recx is returned for this dkey/akey move on */
+        if (number == 0) 
+            continue;
+
+        for (i = 0; i < number; i++) {
+            uint64_t    buf_len = recxs[i].rx_nr;
+            char        buf[buf_len];
+            d_sg_list_t sgl;
+            d_iov_t     iov;
+
+            /* set iod values */
+            (*iod).iod_type  = DAOS_IOD_ARRAY;
+            (*iod).iod_size  = 1;
+            (*iod).iod_nr    = 1;
+            (*iod).iod_recxs = &recxs[i];
+
+            /* set sgl values */
+            sgl.sg_nr     = 1;
+            sgl.sg_nr_out = 0;
+            sgl.sg_iovs   = &iov;
+
+            /* fetch recx values from source */
+            d_iov_set(&iov, buf, buf_len);	
+            rc = daos_obj_fetch(*src_oh, DAOS_TX_NONE, 0, dkey, 1, iod,
+                                &sgl, NULL, NULL);
+
+            /* update fetched recx values and place in destination object */
+            rc = daos_obj_update(*dst_oh, DAOS_TX_NONE, 0, dkey, 1, iod,
+                                 &sgl, NULL);
+        }
+    }
+    return rc;
+}
+
+static int daos_copy_list_keys(daos_handle_t *src_oh,
+                          daos_handle_t *dst_oh)
+{
+    /* loop to enumerate dkeys */
+    daos_anchor_t dkey_anchor = {0}; 
+    int rc;
+    while (!daos_anchor_is_eof(&dkey_anchor)) {
+        d_sg_list_t     sgl;
+        d_iov_t         iov;
+        daos_key_desc_t dkey_kds[ENUM_DESC_NR]       = {0};
+        uint32_t        dkey_number                  = ENUM_DESC_NR;
+        char            dkey_enum_buf[ENUM_DESC_BUF] = {0};
+        char            dkey[ENUM_KEY_BUF]           = {0};
+
+        sgl.sg_nr     = 1;
+        sgl.sg_nr_out = 0;
+        sgl.sg_iovs   = &iov;
+
+        d_iov_set(&iov, dkey_enum_buf, ENUM_DESC_BUF);
+
+        /* get dkeys */
+        rc = daos_obj_list_dkey(*src_oh, DAOS_TX_NONE, &dkey_number, dkey_kds,
+                                &sgl, &dkey_anchor, NULL);
+        if (rc)
+            return -1;       
+
+        /* if no dkeys were returned move on */
+        if (dkey_number == 0)
+            continue;
+
+        char* ptr;
+        int   rc;
+        int   j;
+
+        /* parse out individual dkeys based on key length and numver of dkeys returned */
+        for (ptr = dkey_enum_buf, j = 0; j < dkey_number; j++) {
+
+            /* Print enumerated dkeys */
+            daos_key_t diov;
+            snprintf(dkey, dkey_kds[j].kd_key_len + 1, "%s", ptr);
+            d_iov_set(&diov, (void*)dkey, dkey_kds[j].kd_key_len);
+            ptr += dkey_kds[j].kd_key_len;
+
+            /* loop to enumerate akeys */
+            daos_anchor_t akey_anchor = {0}; 
+            while (!daos_anchor_is_eof(&akey_anchor)) {
+                d_sg_list_t     sgl;
```

Comment:
`sgl` is already defined above in this function.
If we can't reuse it reliably, should we use a different name?